src/Icicle/Sea/FromAvalanche.hs

```diff
@@ -0,0 +1,470 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternGuards #-}
+
+module Icicle.Sea.FromAvalanche (
+    seaOfProgram
+  , accumsOfProgram
+  , outputsOfProgram
+  ) where
+
+import           Icicle.Avalanche.Prim.Flat
+import           Icicle.Avalanche.Program
+import           Icicle.Avalanche.Statement.Statement
+
+import           Icicle.Common.Annot
+import           Icicle.Common.Base
+import           Icicle.Common.Exp
+import qualified Icicle.Common.Exp.Prim.Minimal as M
+import           Icicle.Common.Type
+
+
+import           Icicle.Internal.Pretty
+import qualified Icicle.Internal.Pretty as Pretty
+
+import           P
+
+import qualified Data.List as List
+import           Data.Map (Map)
+import qualified Data.Map as Map
+
+
+------------------------------------------------------------------------
+
+seaOfProgram :: (Show a, Show n, Pretty n, Ord n)
+             => Program (Annot a) n Prim -> Doc
+seaOfProgram program = vsep
+  [ "#include <stdbool.h>"
+  , "#include <stdint.h>"
+  , "#include <math.h>"
+  , ""
+  , "typedef uint64_t iunit_t;"
+  , "typedef uint64_t ibool_t;"
+  , "typedef  int64_t iint_t;"
+  , "typedef   double idouble_t;"
+  , "typedef  int64_t idate_t;"
+  , ""
+  , "typedef const char *ierror_t;"
+  , ""
+  , stateOfProgram program
+  , ""
+  , "static const iunit_t iunit  = 0x1c1c13;"
+  , "static const ibool_t ifalse = 0;"
+  , "static const ibool_t itrue  = 1;"
+  , ""
+  , "#define INLINE __attribute__((always_inline))"
+  , ""
+  , "static idouble_t INLINE iint_extend   (iint_t    x)              { return x; }"
+  , "static iint_t    INLINE iint_add      (iint_t    x, iint_t    y) { return x +  y; }"
```

Comment:
Hmm, we only have two types though, and this makes the pretty printer simpler because everything is prefix.

Did you see the code for `seaOfXPrim` and friends?